app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -565,25 +571,74 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
     for (int i = 0; i < mHandJoints.size(); i++) {
         vrb::Matrix transform = XrPoseToMatrix(mHandJoints[i].pose);
         if (mHandJoints[i].locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT) {
-            float radius = mHandJoints[i].radius;
-            vrb::Matrix scale = vrb::Matrix::Identity().ScaleInPlace(vrb::Vector(radius, radius, radius));
-            transform.PostMultiplyInPlace(scale);
-
             if (renderMode == device::RenderMode::StandAlone)
                 transform.TranslateInPlace(kAverageHeight);
         } else {
             // This effectively hides the joint.
             transform.ScaleInPlace(vrb::Vector(0.0f, 0.0f, 0.0f));
         }
 
-        memcpy(&jointTransforms[i], &transform, sizeof(vrb::Matrix));
+        jointTransforms[i] = transform;
+    }
+
+    // Check whether palm of left hand is facing the head, to show navigate-back button
+    bool leftPalmFacesHead = false;
+    if (mHandeness == OpenXRHandFlags::Left) {
+#if defined(OCULUSVR)
```

Comment:
I know Pico does not properly fill the data for PALM joint but this implementation is not good for extensibility. Instead you should guard with ifdefs PICO's specific wrong behaviour and leave this code unguarded because is suitable for any other platform.

Reply:
Right, I wrote *palm* in the comment but I'm actually using the wrist joint, because Pico doesn't provide the palm joint. I will make the comment consistent, and add guards to use wrist on Pico and Palm in the rest (except Oculus).